ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs

```diff
@@ -149,7 +149,47 @@ public bool TryGetTypeByFullName(string fullName, out Type type)
                 return TypesByFullName.TryGetValue(fullName, out type);
             }
         }
-
+        public void TranlateContentSourceMeta(string publicGitRepoUrl,
+           string publicGitBranch)
+        {
+            foreach (var ns in _nsList)
+            {
+                HandleContentSourceMeta(ns, publicGitBranch, publicGitRepoUrl);
+                foreach (var t in ns.Types)
+                {
+                    HandleContentSourceMeta(t, publicGitBranch, publicGitRepoUrl);
+                    if (t.Members != null)
+                    {
+                        foreach (var m in t.Members)
+                        {
+                            HandleContentSourceMeta(m, publicGitBranch, publicGitRepoUrl);
+                        }
+                        if (t.Overloads != null)
+                        {
+                            foreach (var o in t.Overloads)
+                            {
+                                HandleContentSourceMeta(o, publicGitBranch, publicGitRepoUrl);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        private void HandleContentSourceMeta(ReflectionItem item, string publicGitBranch, string publicGitRepoUrl)
+        {
+            if (item != null && item.Metadata.TryGetValue("contentSourcePath", out object val) && val != null)
+            {
+                var mdPath = val.ToString().Replace("\\", "/");
+                mdPath = mdPath.StartsWith("/") ? mdPath : ("/" + mdPath);
+                item.SourceDetail = new GitSourceDetail()
+                {
+                    Path = mdPath,
+                    RepoBranch = publicGitBranch,
+                    RepoUrl = publicGitRepoUrl
+                };
+                item.Metadata.Remove("contentSourcePath");
+            }
+        }
         public void TranslateSourceLocation(
```

Comment:
we can call `TranlateContentSourceMeta` directly from this method.

Reply:
for "ECMA2YamlConverter.Run()",this method has redundant parameters and logic, and it is only called in the program.Main.